### PR TITLE
Util module and some tests

### PR DESCRIPTION
**Pull request by fedefernandez**
This PR:
- Creates a new sub-module called `util` with some utility logic
- Adds new tests

Please @javipacheco could you review? Thanks

### CODE REVIEW

File: core/src/main/scala/com/fortysevendeg/mvessel/Connection.scala [hunk 1]
```diff
@@ -7,14 +7,13 @@ import java.util.concurrent.Executor
 import com.fortysevendeg.mvessel.logging.{AndroidLogWrapper, LogWrapper}
 import com.fortysevendeg.mvessel.metadata.DatabaseMetaData
 import com.fortysevendeg.mvessel.statement.{PreparedStatement, Statement}
-
-import scala.util.{Failure, Success, Try}
+import com.fortysevendeg.mvessel.util.DatabaseUtils._
 
 class Connection(
```
Comment (javipacheco) [inline core/src/main/scala/com/fortysevendeg/mvessel/Connection.scala:12]: It's possible that you can use `case class` here, and all your parameters will be `val`s 

I don't know how you are using this class. It's only a comment for if you want to try it

File: core/src/main/scala/com/fortysevendeg/mvessel/Driver.scala [hunk 2]
```diff
@@ -6,11 +6,12 @@ import java.util.logging.Logger
...
 import com.fortysevendeg.mvessel.Connection._
 import com.fortysevendeg.mvessel.Driver._
-import com.fortysevendeg.mvessel.util.ConnectionStringParser._
+import com.fortysevendeg.mvessel.util.ConnectionStringParser
+import com.fortysevendeg.mvessel.util.DatabaseUtils.WrapSQLException
 
 import scala.util.{Failure, Try}
 
-class Driver extends SQLDriver {
+class Driver extends SQLDriver with ConnectionStringParser {
```
Comment (javipacheco) [inline core/src/main/scala/com/fortysevendeg/mvessel/Driver.scala]: Every trait in a new line?

File: core/src/test/scala/com/fortysevendeg/mvessel/DriverSpec.scala [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    "return an empty array" in new WithConnectionValues {
+      driver.getPropertyInfo(validUrl, properties) must beEmpty
+    }
+
+  }
+
+  "getMinorVersion" should {
+
+    "return 0" in new WithConnectionValues {
+      driver.getMinorVersion shouldEqual 0
```
Comment (javipacheco) [inline core/src/test/scala/com/fortysevendeg/mvessel/DriverSpec.scala]: Magic number?

File: core/src/test/scala/com/fortysevendeg/mvessel/DriverSpec.scala [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+  }
+
+  "getMajorVersion" should {
+
+    "return 1" in new WithConnectionValues {
+      driver.getMajorVersion shouldEqual 1
```
Comment (javipacheco) [inline core/src/test/scala/com/fortysevendeg/mvessel/DriverSpec.scala]: Magic number?

**Comment by javipacheco** [comment]
Minor changes only if you consider it necessary. LGTM!

File: core/src/main/scala/com/fortysevendeg/mvessel/Connection.scala (hunk 1 above)
Comment (fedefernandez) [inline core/src/main/scala/com/fortysevendeg/mvessel/Connection.scala:12]: Yeah, my only concern is that the `Connection` class is an implementation for the java.sql.Connection. I think it's better to let a pure class. thanks

File: core/src/main/scala/com/fortysevendeg/mvessel/Driver.scala [hunk 5]
```diff
@@ -71,12 +71,11 @@ class Driver extends SQLDriver {
 
 object Driver {
 
-  Try {
-    java.sql.DriverManager.registerDriver(new Driver)
-  } match {
-    case Failure(e) => e.printStackTrace()
-    case _ =>
-  }
+  def registerDriver() =
```
Comment (47degci) [inline core/src/main/scala/com/fortysevendeg/mvessel/Driver.scala]: Codacy Issue found: [You should not name methods after their object](https://www.codacy.com/app/47deg/mvessel/file/848926616/issues/source?bid=2442814&fileBranchId=2811084#l74)

File: util/src/main/scala/com/fortysevendeg/mvessel/util/StructureControlProcessor.scala [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    @tailrec
+    def loop(seq: Seq[T]): Seq[T] = {
+      if (untilTrue(seq.size) && processor.move(c)) loop(seq :+ process(c))
+      else {
+        processor.close(c)
+        seq
+      }
+    }
+
+    Try(loop(Seq.empty)) match {
```
Comment (47degci) [inline util/src/main/scala/com/fortysevendeg/mvessel/util/StructureControlProcessor.scala:50]: Codacy Issue found: [Default case not specified.](https://www.codacy.com/app/47deg/mvessel/file/848926663/issues/source?bid=2442814&fileBranchId=2811084#l50)

File: util/src/main/scala/com/fortysevendeg/mvessel/util/DatabaseUtils.scala [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        throw new SQLException(message)
+    }
+
+  }
+
+  object WithStatement {
+
+    def apply[T](f: Statement => T)(implicit connection: Connection): T = {
+      val statement = connection.createStatement()
+      Try(f(statement)) match {
```
Comment (47degci) [inline util/src/main/scala/com/fortysevendeg/mvessel/util/DatabaseUtils.scala:26]: Codacy Issue found: [Default case not specified.](https://www.codacy.com/app/47deg/mvessel/file/848926658/issues/source?bid=2442814&fileBranchId=2811084#l26)

File: util/src/main/scala/com/fortysevendeg/mvessel/util/DatabaseUtils.scala [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import java.sql.{SQLException, Connection, Statement}
+import scala.util.{Failure, Success, Try}
+
+object DatabaseUtils {
+
+  object WrapSQLException {
+
+    def apply[D, T](option: Option[D], message: String)(f: D => T) = option match {
+      case Some(db) =>
+        Try(f(db)) match {
```
Comment (47degci) [inline util/src/main/scala/com/fortysevendeg/mvessel/util/DatabaseUtils.scala:12]: Codacy Issue found: [Default case not specified.](https://www.codacy.com/app/47deg/mvessel/file/848926658/issues/source?bid=2442814&fileBranchId=2811084#l12)

**Pull request merged**